src/Microsoft.Extensions.DependencyModel/CompilationLibrary.cs

```diff
@@ -51,8 +51,32 @@ public class CompilationLibrary : Library
 
         public IEnumerable<string> ResolveReferencePaths()
         {
+            return ResolveReferencePaths(DefaultResolver);
+        }
+
+        public IEnumerable<string> ResolveReferencePaths(params ICompilationAssemblyResolver[] additionalResolvers)
```

Comment:
I'm a bit concerned by performance of doing this per `CompilationLibrary`. Also resolvers happen throw in some cases (https://github.com/dotnet/core-setup/blob/master/src/Microsoft.Extensions.DependencyModel/Resolution/AppBaseCompilationAssemblyResolver.cs#L102) so having additional resolvers last might not solve everything.